src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -442,15 +442,21 @@ void WorldSession::HandleLogoutCancelOpcode(WorldPacket& /*recvData*/)
     }
 }
 
-void WorldSession::HandleTogglePvP(WorldPacket& recvData)
+void WorldSession::HandleTogglePvP(WorldPackets::Misc::TogglePvP& togglePvP)
 {
     // this opcode can be used in two ways: Either set explicit new status or toggle old status
-    if (recvData.size() == 1)
+    if (togglePvP.HasPvPStatus())
     {
-        bool newPvPStatus;
-        recvData >> newPvPStatus;
-        GetPlayer()->ApplyModFlag(PLAYER_FLAGS, PLAYER_FLAGS_IN_PVP, newPvPStatus);
```

Comment:
Is ApplyModFlag() not working with the new packet structure ?

Reply:
It is, it was changed in master for a different reason (incompatible with 8.1+ updatefield format) and since this is a port from master branch, it has the `if ()` separation too

Reply:
I changed this because in my opinion logic has become easier to understand than before

Reply:
I think I would prefer to keep ApplyModFlag as a helper for setting/removing flags based on bool variable.
Maybe it's not so easy to understand because it's not used that much around the code.